File: hw/src/main/java/com/sumian/app/improve/widget/SleepStateDurationView.java
package com.sumian.app.improve.widget;

import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.Gravity;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.sumian.app.R;
import com.sumian.app.common.util.TimeUtil;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by sm
 * on 2018/3/8.
 * desc:睡眠报告中,睡眠时长统计容器
 */

public class SleepStateDurationView extends LinearLayout {

    @BindView(R.id.tv_label)
    TextView mTvLabel;

    @BindView(R.id.tv_sleep_duration_percent)
    SleepPercentTextView mTvSleepDurationPercent;

    @BindView(R.id.tv_sleep_duration_count)
    TextView mTvSleepDurationCount;

    public SleepStateDurationView(Context context) {
        this(context, null);
    }

    public SleepStateDurationView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public SleepStateDurationView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setGravity(Gravity.CENTER);
        setOrientation(VERTICAL);
        initView(context);
    }

    private void initView(Context context) {
        ButterKnife.bind(inflate(context, R.layout.hw_lay_sleep_state_duration_view, this));
    }

    public void setData(String label, int duration, int percent) {
        mTvLabel.setText(label);
        mTvSleepDurationCount.setText(TimeUtil.formatSleepDurationText(getContext(),duration));
        mTvSleepDurationPercent.setPercent(percent);
    }
}

File: SumianDeviceDemo/src/main/java/com/sumian/devicedemo/sleepdata/widget/SleepStateDurationView.kt
package com.sumian.sd.buz.report.widget

import android.content.Context
import android.util.AttributeSet
import android.view.Gravity
import android.view.View
import android.widget.LinearLayout
import com.sumian.devicedemo.R
import kotlinx.android.synthetic.main.hw_lay_sleep_state_duration_view.view.*

/**
 * Created by sm
 * on 2018/3/8.
 * desc:睡眠报告中,睡眠时长统计容器
 */

class SleepStateDurationView @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet? = null,
        defStyleAttr: Int = 0
) : LinearLayout(context, attrs, defStyleAttr) {

    init {
        gravity = Gravity.CENTER
        orientation = LinearLayout.VERTICAL
        initView(context)
    }

    private fun initView(context: Context) {
        View.inflate(context, R.layout.hw_lay_sleep_state_duration_view, this)
    }

    fun setData(label: String, duration: Int, percent: Int) {
        tv_label.text = label
        tv_sleep_duration_count.setDuration(duration)
        tv_sleep_duration_percent.setPercent(percent)
    }
}
